website/src/docs/hotchocolate/execution-engine/field-middleware.md

```diff
@@ -0,0 +1,252 @@
+---
+title: Field middleware
+---
+
+TODO introduce field middleware
+
+# Execution order
+
+TODO diagram, data middleware, why order matters
+
+# Definition
+
+Field middleware can be defined either as a delegate or as a separate type. In both cases we gain access to a `FieldDelegate` (`next`) and the `IMiddlewareContext` (`context`).
+
+The `IMiddlewareContext` implements the `IResolverContext` interface so you can use all of the `IResolverContext` APIs in your middleware, similarly to how you would use them in your resolver code.
+
+By awaiting the `FieldDelegate` we are waiting on all other field middleware that might come after the current middleware as well as the actual field resolver, computing the result of the field.
+
+## Field middleware delegate

```

Comment:
Shall we also describe the request middleware for extending the execution as a whole? If you want I can write something up on that, I already have most of the diagrams for that.

Reply:
I assume by request middleware you mean the pipeline stuff, i.e. `UseRequest`? I wanted to do a different document on that.

Reply:
Yes